software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessDriver.java

```diff
@@ -108,23 +108,24 @@ public void start() {
         }
         if (!skipStart) {
             DynamicTasks.queue("install", new Runnable() { public void run() {
-
                 Optional<Boolean> locationInstalled = Optional.fromNullable(getLocation().getConfig(BrooklynConfigKeys.SKIP_ENTITY_INSTALLATION));
                 Optional<Boolean> entityInstalled = Optional.fromNullable(entity.getConfig(BrooklynConfigKeys.SKIP_ENTITY_INSTALLATION));
-                boolean skipInstall = locationInstalled.or(entityInstalled).or(false);
-                if (!skipInstall) {
-                    DynamicTasks.queue("copy-pre-install-resources", new Runnable() { public void run() {
-                        waitForConfigKey(BrooklynConfigKeys.PRE_INSTALL_RESOURCES_LATCH);
-                        copyPreInstallResources();
-                    }});
 
-                    DynamicTasks.queue("pre-install", new Runnable() { public void run() {
-                        preInstall();
-                    }});
+                DynamicTasks.queue("copy-pre-install-resources", new Runnable() { public void run() {
+                    waitForConfigKey(BrooklynConfigKeys.PRE_INSTALL_RESOURCES_LATCH);
+                    copyPreInstallResources();
+                }});
 
-                    DynamicTasks.queue("pre-install-command", new Runnable() { public void run() {
-                        runPreInstallCommand();
-                    }});
+                DynamicTasks.queue("pre-install", new Runnable() { public void run() {
+                    preInstall();
+                }});
+
+                DynamicTasks.queue("pre-install-command", new Runnable() { public void run() {
+                    runPreInstallCommand();
+                }});
+
+                boolean skipInstall = locationInstalled.or(entityInstalled).or(false);
+                if (!skipInstall) {
```

Comment:
Just to state the cases we need to test or to have automated test:

1. Install multiple apps on a BYON location
1. Perform a restart on a BYON location
1. Perform a restart with machineStop: NEVER/(or false)